baseplate/context/__init__.py

```diff
@@ -53,3 +53,7 @@ def __init__(self, name, context_factory):
     def on_server_span_created(self, context, server_span):
         context_attr = self.context_factory.make_object_for_context(self.name, server_span)
         setattr(context, self.name, context_attr)
+
+    def on_local_span_created(self, context, span, component_name):
+        context_attr = self.context_factory.make_object_for_context(self.name, span)
```

Comment:
It looks like this would probably confuse the SQLAlchemySessionContextFactory: https://github.com/reddit/baseplate/blob/master/baseplate/context/sqlalchemy.py#L102 as it'd close the session at the end of a local span before the server span had finished.

We probably need to look at the type of the span (ew?) and only do the session closing on server spans or something else smarter there.